src/core/border_router/routing_manager.cpp

```diff
@@ -3324,14 +3329,76 @@ void RoutingManager::RsSender::HandleTimer(void)
 }
 
 #if OPENTHREAD_CONFIG_BORDER_ROUTING_DHCP6_PD_ENABLE
+const char *RoutingManager::PdPrefixManager::StateToString(Dhcp6PdState aState)
+{
+    static const char *const kStateStrings[] = {
+        "Disabled", // (0) kDisabled
+        "Stopped",  // (1) kStopped
+        "Running",  // (2) kRunning
+    };
+
+    static_assert(0 == kDhcp6PdStateDisabled, "kDhcp6PdStateDisabled value is incorrect");
+    static_assert(1 == kDhcp6PdStateStopped, "kDhcp6PdStateStopped value is incorrect");
+    static_assert(2 == kDhcp6PdStateRunning, "kDhcp6PdStateRunning value is incorrect");
+
+    return kStateStrings[aState];
+}
+
 RoutingManager::PdPrefixManager::PdPrefixManager(Instance &aInstance)
     : InstanceLocator(aInstance)
     , mEnabled(false)
+    , mIsRunning(false)
     , mTimer(aInstance)
 {
     mPrefix.Clear();
 }
 
+void RoutingManager::PdPrefixManager::StartStop(bool aStart)
```

Comment:
nit: rename to `SetEnabled()`?

Reply:
We already have `SetEnabled()` to disable / enable the whole PdPrefixManager (regardless the network data and infralink status). Just like `RoutingManager::SetEnabled()`

`RoutingManager` has `Start` and `Stop`, however, the content of `Start` and `Stop` differ a lot.

I have tried to have separate `Start()` and `Stop()` for `PdPrefixManager`, but it does not make sense to me since `Start()` and `Stop()` only differ in a single bit.

So I make the `Start()` and `Stop()` to be an alias for `StartStop(true)` and `StartStop(false)` for readability and use a centralized `StartStop` to avoid potential bugs by typos.